account_payment_order_to_voucher/models/payment_order.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+#
+#   See __openerp__.py about license
+#
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning
+
+
+class PaymentOrder(models.Model):
+    _inherit = 'payment.order'
+    voucher_ids = fields.Many2many(
+        'account.voucher', string='Vouchers', readonly=True)
+
+    def get_lines_by_partner(self, order):
+        lines_by_partner = {}
+        if order.voucher_ids:
+            raise Warning(
+                _("Payment order %s already has vouchers")
+                % order.reference)
+        if order.state != 'done':
+            raise Warning(
+                _("Payment order %s is not in 'done' state")
+                % order.reference)
+        for line in order.line_ids:
+            if line.partner_id.id not in lines_by_partner:
+                lines_by_partner[line.partner_id.id] = []
+            lines_by_partner[line.partner_id.id].append(line)
+        return lines_by_partner
+
+    def _compute_lines_total(self, payment_lines):
+        total = 0.0
+        for line in payment_lines:
+            total += line.amount_currency
+        return total
+
+    def _get_currency_id(self, payment_lines):
+        currency_ids = [l.currency.id for l in payment_lines]
```

Comment:
In this case I would keep the current, because 1 line would become 2:
```
        currencies = payment_lines.mapped('currency')
        currency_ids = currencies.mapped('id')
```